tests/ranfunc/gitopsztp/tests/gitopsztp_policies_app.go

```diff
@@ -0,0 +1,138 @@
+			if !ranfunchelper.IsVersionStringInRange(
+				ranfunchelper.ZtpVersion,
+				"4.10",
+				"",
+			) {
+				Skip(fmt.Sprintf(
+					"unable to run test on ztp version '%s' as it is less than minimum '%s",
+					ranfunchelper.ZtpVersion,
+					"4.10",
+				))
+			}
+		})
+
+		// Collecting and storing default Git values of policies app before the test
+		By("Collecting original Git values of policies app", func() {
+			defaultGitRepo, defaultGitBranch, defaultGitPath, err = gitopsztphelper.GetGitDetailsFromArgocd(
+				gitopsztpparams.ArgocdPoliciesAppName, ranfuncparams.OpenshiftGitops)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	AfterEach(func() {
+		// Reset the policies app back to default after each test
+		By("Resetting the policies app back to the original settings", func() {
+			err := gitopsztphelper.SetGitDetailsInArgocd(
+				defaultGitRepo, defaultGitBranch, defaultGitPath,
+				gitopsztpparams.ArgocdPoliciesAppName,
+				true,
+				false,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("Apply and Validate custom source CRs on the du policies", Label("ztp-custom-source-crs"), func() {
+
+		var (
+			// customSourceCrPolicyName = "custom-source-cr-policy-config" // enforce policy
+			testCrName       = "custom-source-cr"
+			testNs           = "default"
+			crServiceAccount = serviceaccount.NewBuilder(ranfuncinittools.SpokeAPIClient, testCrName, testNs)
+		)
+
+		It("verifies new CR kind that does not exist in ztp "+
+			"container image can be created via custom source-cr", polarion.ID("61978"), func() {
+			// The ztp test data is stored in a nested directory within the ztp repo
+			testGitPath := gitopsztphelper.JoinGitPaths(
+				[]string{
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Path,
+					"ztp-test/custom-source-crs/new-cr",
+				},
+			)
+
+			By("Checking if the git path exists", func() {
+				if !gitopsztphelper.DoesGitPathExist(
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Repo,
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Branch,
+					testGitPath+"/kustomization.yaml",
+				) {
+					Skip(fmt.Sprintf("git path '%s' could not be found", testGitPath))
+				}
+			})
+
+			By("Checking Service Account custom resource does NOT exist on spoke", func() {
+				err := crServiceAccount.Exists()
+				Expect(err).To(BeFalse())
+			})
+
+			By("Updating the Argocd app", func() {
+				// Update the Argo app to point to the new test kustomization
+				err := gitopsztphelper.SetGitDetailsInArgocd(
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Repo,
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Branch,
+					testGitPath,
+					gitopsztpparams.ArgocdPoliciesAppName,
+					true,
+					true,
+				)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Checking new Service Account custom resource created and applied to spoke", func() {
+				err := crServiceAccount.Exists()
```

Comment:
please rename err here, as it is not returning an error.

Reply:
Ack,done.